source/common/filter/tcp_proxy.h

```diff
@@ -30,7 +30,11 @@ namespace Filter {
   COUNTER(downstream_cx_tx_bytes_total)                                                            \
   GAUGE  (downstream_cx_tx_bytes_buffered)                                                         \
   COUNTER(downstream_cx_total)                                                                     \
-  COUNTER(downstream_cx_no_route)
+  COUNTER(downstream_cx_no_route)                                                                  \
+  COUNTER(upstream_pause_reading)                                                                  \
```

Comment:
Actually I have a one more questions on this (sorry for the back and forth).  

Added upstream_pause_reading_total to cluster stats in my local branch but then it only gets watermark enables/disables.  Any preference for a rename for flow_control_[paused|resumed]_reads vs tracking stats in readDisable on the connection?

Given there's already some readEnabling/Disabling going on on the http side I'm not sure if you'd prefer to track the flow control changes separately or if tracking all enables/disables is more useful.